### ISSUE TITLE
Add more user-friendly handling for failed network requests on app start

### ISSUE THREAD

**Issue by otech47** [lightninglabs/lightning-app#1192]
### Description

The Android app crashes on a fresh open when there is no network connection. There should be some kind of error handling with a notice to the user that network requests are failing.

### Steps to reproduce the behavior

- Kill/force-close app
- Turn off WiFi and data
- Open the Lightning app
- App crashes after a screen transition (does not reach enter PIN screen)

### Expected behavior

There should be a simple informative notice shown to the user: "Please check your network connection". Ideally a background listener is implemented to detect network changes and reattempt the network requests to initialize the app. Maybe a "Try Again" button would be simpler and more user-friendly.

### Actual behavior

App crashes.

### Screenshot and/or logs if applicable

Sent via Android reporting. You should see it in the Play Developer Console

**Comment by jtayrone** [comment]
In my case it's almost like this, because after the PIN screen after a few seconds the APP simply stops 
![APP simply stops ](https://user-images.githubusercontent.com/34479816/59880033-733e3900-9382-11e9-8aba-66040bcbc256.gif)
working.

**Comment by otech47** [comment]
@jtayrone you seem to have WiFi enabled though? I turned it completely off. Could be the same error but the request at least goes through WiFi and fails eventually in your case?

**Comment by jtayrone** [comment]
Exact, Wi-Fi enabled, but with Wifi turned off without any network the APP failed to initialize.
